enterprise/cmd/frontend/internal/registry/allow.go

```diff
@@ -23,19 +23,40 @@ func init() {
 		return false
 	}
 
+	frontendregistry.IsRemoteExtensionPublisherAllowed = func(x *registry.Extension) bool {
+		if getAllowOnlySourcegraphAuthoredExtensionsFromSiteConfig() {
+			return isSourcegraphAuthoredExtension(x)
+		}
+
+		return true
+	}
+
 	frontendregistry.FilterRemoteExtensions = func(extensions []*registry.Extension) []*registry.Extension {
-		allowedExtensions := getAllowedExtensionsFromSiteConfig()
-		if allowedExtensions == nil {
-			// Default is to allow all extensions.
-			return extensions
+		var extensionsWithAllowedPublisher []*registry.Extension
+
+		if getAllowOnlySourcegraphAuthoredExtensionsFromSiteConfig() {
+			for _, x := range extensions {
+				if isSourcegraphAuthoredExtension(x) {
+					extensionsWithAllowedPublisher = append(extensionsWithAllowedPublisher, x)
+				}
+			}
+		} else {
+			// If allow only Sourcegraph extensions feature is not enabled, all remote extensions are allowed.
+			extensionsWithAllowedPublisher = extensions
+		}
+
+		allowedExtensionIDs := getAllowedExtensionsFromSiteConfig()
+
+		if allowedExtensionIDs == nil {
```

Comment:
```suggestion
		if len(allowedExtensionIDs) == 0 {
```

It is better and more robust to check for length of a slice than the `nil`.

Reply:
@unknwon, I wonder if it covers the case when [`getAllowedExtensionsFromSiteConfig` returns `nil`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@taras-yemets/add-allow-only-sourcegraph-extensions-setting/-/blob/enterprise/cmd/frontend/internal/registry/allow.go?L68-79)?
Sorry, I'm new to Go and I may be asking silly questions 🙂

Reply:
It does, that's why `len` is better 😁

Reply:
I think I get why only `nil` check is here.
When allowed extensions are an empty slice it looks like [we want to return an empty slice too](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/enterprise/cmd/frontend/internal/registry/allow_test.go?L82)(no remote extensions are allowed), but when it's `nil` we just [return all the extensions](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/enterprise/cmd/frontend/internal/registry/allow_test.go?L79) (all are allowed).

But if the empty slice means that no remote extensions are allowed, I wonder how is it different from [the `false` value](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/schema/site.schema.json?L1022%3A11=).

Reply:
> But if the empty slice means that no remote extensions are allowed, I wonder how is it different from [the false value](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/schema/site.schema.json?L1022%3A11=).

Ah, `false` is a possible value of the `remoteRegistry` field. Ignore this part, please. Sorry for the confusion.